sponsorship_compassion/migrations/10.0.1.0.6/post-migration.py

```diff
@@ -0,0 +1,40 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Copyright (C) 2018 Compassion CH (http://www.compassion.ch)
+#    @author: Quentin Gigon <gigon.quentin@gmail.com>
+#
+#    The licence is in the file __manifest__.py
+#
+##############################################################################
+from odoo import fields
+from openupgradelib import openupgrade
+
+import random
+import dateutil.parser
+
+
+@openupgrade.migrate(use_env=True)
+def migrate(env, version):
+    if not version:
+        return
+
+    end_date_limit = dateutil.parser.parse("2016-12-13").date()
+
+    old_sponsorships = env['recurring.contract'].search([
+        ('end_date', '<=', fields.Date.to_string(end_date_limit)),
```

Comment:
Replace this condition by `('type', 'like', 'S')` to filter only sponsorships. The end_date makes no sense to filter after reflection.